### PR TITLE
added scanning line in Sonar View Widget

### PR DESCRIPTION
**Pull request by romulogcerqueira**
added scanning line in Sonar View Widget

### CODE REVIEW

**Comment by goldhoorn** [comment]
Can someone this double-check whether it's working or not?
@D-Alex or maybe @fzachert (if you are workin with sonars?)

@romulogcerqueira: can you please remove the commented out lines you added to keep the code clean?

File: src/sonar_view/SonarView.cc [hunk 1]
```diff
@@ -111,8 +111,12 @@ void SonarView::setSonarScan(const char *data_, int size, double angle, double t
 		}
 	}
 	lastBearing = bearing;
-        addImage(img);
+
+	addImage(output);
+//	        addImage(img);
```
Comment (goldhoorn) [inline src/sonar_view/SonarView.cc]: REmove comments like this

File: src/sonar_view/SonarView.cc [hunk 2]
```diff
@@ -111,8 +111,12 @@ void SonarView::setSonarScan(const char *data_, int size, double angle, double t
...
+
```
Comment (goldhoorn) [inline src/sonar_view/SonarView.cc]: Remove empty lines like this

File: src/sonar_view/SonarView.h [hunk 3]
```diff
@@ -20,12 +20,12 @@ class QDESIGNER_WIDGET_EXPORT SonarView : public ImageViewOld
 
 public slots:    
 	void setOpenGL(bool flag);
-        void setSonarScan(const char *data, int size, double angle,double timeBetweenBins ,bool fromBearing=false);
+    void setSonarScan(const char *data, int size, double angle,double timeBetweenBins ,bool fromBearing=false);
```
Comment (goldhoorn) [inline src/sonar_view/SonarView.h]: Please do not reformat code without a change

If you unhappy (or the indentation is wrong in general) please create a separate commit out of this

File: src/sonar_view/SonarView.cc (hunk 1 above)
Comment (romulogcerqueira) [inline src/sonar_view/SonarView.cc]: Done!

File: src/sonar_view/SonarView.cc (hunk 2 above)
Comment (romulogcerqueira) [inline src/sonar_view/SonarView.cc]: Done!

File: src/sonar_view/SonarView.h (hunk 3 above)
Comment (romulogcerqueira) [inline src/sonar_view/SonarView.h]: Ok. Done!

**Comment by romulogcerqueira** [comment]
Hi @goldhoorn,

the current SonarView has some flickers in the image and it is not optimized. I discussed with @doudou to create a new clean sonar viewer using OpenCV or OSG.

Additionally, the SonarWidget is not working for polar coordinates. At this moment, I am using @fzachert fork (https://github.com/fzachert/gui-rock_widget_collection) that correct this.

**Comment by doudou** [comment]
> the current SonarView has some flickers in the image and it is not optimized. I discussed with @doudou to create a new clean sonar viewer using OpenCV or OSG.

No you did not. I really don't see why OCV or OSG would be needed for the job. Moreover, the SonarWidget seems to be where the work is being done these days. With the new sonar data type we should probably try to focus on having a single widget and get rid of SonarView altogether, or ?

**Comment by goldhoorn** [comment]
pong how is the state of this?

**Comment by romulogcerqueira** [comment]
Hi @goldhoorn ,

I have done some fixes in the SonarWidget and I will commit them soon in another branch.

For now, I will close this PR.

Thanks.

**Pull request closed without merging**